src/eos-speedwagon.c

```diff
@@ -43,6 +43,78 @@ typedef struct {
 GType eos_speedwagon_window_get_type (void) G_GNUC_CONST;
 G_DEFINE_TYPE (EosSpeedwagonWindow, eos_speedwagon_window, GTK_TYPE_APPLICATION_WINDOW)
 
+static gchar *
+get_splash_background_path (GDesktopAppInfo *app_info)
+{
+  gchar *bg_path = NULL;
+
+  if (g_desktop_app_info_has_key (app_info,
+                                  SPLASH_BACKGROUND_DESKTOP_KEY))
+    {
+      gchar *bg_name =
+        g_desktop_app_info_get_string (app_info,
+                                       SPLASH_BACKGROUND_DESKTOP_KEY);
+
+      /* rewrite absolute paths */
+      if (g_path_is_absolute (bg_name))
```

Comment:
I was thinking of that as well. I think checking whether it exists would
also fix my bundle issue, but I'd still rather just wait if given an option.

On Fri, Nov 13, 2015, 10:06 AM Dan Nicholson <notifications@github.com>
wrote:

> In src/eos-speedwagon.c
> <https://github.com/endlessm/eos-speedwagon/pull/4#discussion_r44813462>:
>
> > @@ -43,6 +43,78 @@ typedef struct {
> >  GType eos_speedwagon_window_get_type (void) G_GNUC_CONST;
> >  G_DEFINE_TYPE (EosSpeedwagonWindow, eos_speedwagon_window, GTK_TYPE_APPLICATION_WINDOW)
> >
> > +static gchar *
> > +get_splash_background_path (GDesktopAppInfo *app_info)
> > +{
> > +  gchar *bg_path = NULL;
> > +
> > +  if (g_desktop_app_info_has_key (app_info,
> > +                                  SPLASH_BACKGROUND_DESKTOP_KEY))
> > +    {
> > +      gchar *bg_name =
> > +        g_desktop_app_info_get_string (app_info,
> > +                                       SPLASH_BACKGROUND_DESKTOP_KEY);
> > +
> > +      /* rewrite absolute paths */
> > +      if (g_path_is_absolute (bg_name))
>
> Instead of always rewriting absolute paths, maybe it would be a good idea
> to check first if it exists and return early if it does. I don't see us
> really needing that functionality, but it seems more robust if you're in a
> situation where speedwagon is updated but shell-content isn't. E.g.
>
> if (g_path_is_absolute (bg_name))
>   {
>     if (g_file_test (bg_name, G_FILE_TEST_EXISTS))
>       return g_strdup (bg_name);
>     /* continue with basename logic */
>   }
>
> Although, I guess this is what @magcius <https://github.com/magcius> was
> objecting to.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/endlessm/eos-speedwagon/pull/4/files#r44813462>.
>